test/unit_tests/api/test_settings.py

```diff
@@ -43,3 +43,20 @@ def test_set_engine_status(test_client):
     assert response.json()["running_processes"] == 0
     assert response.json()["global_lock"] is False
     assert response.json()["global_status"] == "RUNNING"
+
+
+def test_reset_search_index(test_client, generic_subscription_1, generic_subscription_2):
+    # Initially no subscriptions found in index
+    response = test_client.get("/api/subscriptions/search?query=status:active")
+    assert response.status_code == HTTPStatus.OK
+    subscriptions = response.json()
+    assert len(subscriptions) == 0
+
+    # Refresh index
+    response = test_client.post("/api/settings/search-index/reset")
+    assert response.status_code == HTTPStatus.OK
+
+    # Subscriptions are now found
+    response = test_client.get("/api/subscriptions/search?query=status:active")
+    subscriptions = response.json()
+    assert {sub["subscription_id"] for sub in subscriptions} == {generic_subscription_1, generic_subscription_2}
```

Comment:
I'd rather not have for loops in tests. Better to have 2 explicit asserts.

Reply:
It's a single collection assert. Two separate assertions for the two ids would not be the same. (I want exactly those two). Maybe the following is clearer

```suggestion
    subscription_ids  = {sub["subscription_id"] for sub in subscriptions}
    assert subscription_ids == {generic_subscription_1, generic_subscription_2}
```

Reply:
Them maybe something like:

```python
assert len(subscriptions) == 2
assert subscriptions[0]["subscription_id"] == generic_subscription_1
assert subscriptions[1]["subscription_id"] == generic_subscription_2
```

Reply:
Now it is order dependent. I think the set equality check communicates the intent most clearly. What is the problem with the original solution exactly?

Reply:
Mental overhead instead of simple asserts. Maybe not a big deal in this case, but we have loads of tests that contain code (by which I mean for-loops, conditionals, etc.) in several places. I want to avoid testing the test code, having to execute code in my head to understand what we are actually testing, etc.

Of ordering is not important, you could also use:

```
actual_subscriptions = {subscriptions[0]["subscription_id"], subscriptions[1]["subscription_id"]}
assert generic_subscription_1 in actual_subscriptions
assert generic_subscription_2 in actual_subscriptions
```